### PR TITLE
Add deployment script

### PR DESCRIPTION
**Pull request by fedgiac**
Adds a standard way to deploy a contract and verify its code on the chain's block explorer. See readme changes for details.

The main purpose of the deployment script is providing and validating the expected deployment parameters without them being user input at the time of deployment.

Deployments are non-deterministic. (Having deterministic deployments is easy with Foundry, however the wrapped token address is different on every chain, so it would be a bit of (imho) unnecessary extra work to make the contract addresses the same on every chain. 
The console output contains useful information like txhash and gas required.

I plan to replace `--rpc-url https://long.form.rinkeby.url` with `--rpc-url rinkeby` as soon as `foundry-rs/foundry/pull/3116` is released.

### Test plan

First, test dry run on all supported chains and verify that there is no error:

```sh
export INFURA_KEY=...
forge script script/Deploy.sol --rpc-url "https://mainnet.infura.io/v3/$INFURA_KEY" -vvvv
forge script script/Deploy.sol --rpc-url "https://rinkeby.infura.io/v3/$INFURA_KEY" -vvvv
forge script script/Deploy.sol --rpc-url "https://goerli.infura.io/v3/$INFURA_KEY" -vvvv
forge script script/Deploy.sol --rpc-url "https://rpc.gnosischain.com" -vvvv
```

Then, try to deploy a contract on Rinkeby:
```sh
forge script script/Deploy.sol --rpc-url "https://rinkeby.infura.io/v3/$INFURA_KEY" -vvvv --private-key "$PK" --broadcast
```

<details><summary>output</summary>

```
$ forge script script/Deploy.sol --rpc-url "https://rinkeby.infura.io/v3/$INFURA_KEY" -vvvv --private-key "$PK" --broadcast
[⠆] Compiling...
No files changed, compilation skipped
Traces:
  [1170880] Deploy::run() 
    ├─ [0] VM::startBroadcast() 
    │   └─ ← ()
    ├─ [272] GPv2Settlement::domainSeparator() [staticcall]
    │   └─ ← 0x30208afd16dc027408ef4f1724ce2b47e7a95b39820fb5910ac6708953f96ea1
    ├─ [3264] WETH9::symbol() [staticcall]
    │   └─ ← "WETH"
    ├─ [1033332] → new CoWSwapEthFlow@"0xaa88…237a"
    │   └─ ← 5047 bytes of code
    ├─ [0] VM::stopBroadcast() 
    │   └─ ← ()
    └─ ← ()


Script ran successfully.
==========================
Simulated On-chain Traces:

  [1163008] → new CoWSwapEthFlow@"0xaa88…237a"
    └─ ← 5047 bytes of code


==========================

Estimated total gas used for script: 1511910

Estimated amount required: 0.0045357300302382 ETH

==========================

###
Finding wallets for all the necessary addresses...
##
Sending transactions [0 - 0].
⠁ [00:00:00] [################################################################################################################################################################################] 1/1 txes (0.0s)
Transactions saved to: /app/ethflowcontract/broadcast/Deploy.sol/4/run-latest.json

##
Waiting for receipts.
⠉ [00:00:14] [#######################################################################################################################################################################] 1/1 receipts (0.0s)
#####
✅ Hash: 0x0b4972db4162c9418525c4e2db88cc627144ac34e960fb880544ecc5b7e7aa7f
Contract Address: 0xaa88a3b70633c9746c5d258f402dae041e7f237a
Block: 11339402
Paid: 0.003489024010467072 ETH (1163008 gas * 3.000000009 gwei)


Transactions saved to: /app/ethflowcontract/broadcast/Deploy.sol/4/run-latest.json



==========================

ONCHAIN EXECUTION COMPLETE & SUCCESSFUL. Transaction receipts written to "/app/ethflowcontract/broadcast/Deploy.sol/4/run-latest.json"

Transactions saved to: /app/ethflowcontract/broadcast/Deploy.sol/4/run-latest.json
```
</details>

Finally, try to verify the deployment:

```sh
forge script script/Deploy.sol --rpc-url "https://rinkeby.infura.io/v3/$INFURA_KEY" -vvvv --verify
```

<details><summary>output</summary>

```
$ forge script script/Deploy.sol --rpc-url "https://rinkeby.infura.io/v3/$INFURA_KEY" -vvvv --verify
[⠆] Compiling...
No files changed, compilation skipped
##
Start verification for (1) contracts

Submitting verification for [src/CoWSwapEthFlow.sol:CoWSwapEthFlow] Ok("0xaA88a3B70633c9746c5D258F402daE041e7f237A").
Submitted contract for verification:
        Response: `OK`
        GUID: `mc2szh8wpyp831nuj3n7fqnub6urx9zyfsxybp5izzuupvtbnf`
        URL:
        https://rinkeby.etherscan.io/address/KEY
Waiting for verification result...
Contract successfully verified
All (1) contracts were verified!

Transactions saved to: /app/ethflowcontract/broadcast/Deploy.sol/4/run-latest.json
```
</details>

See result [on Etherscan](https://rinkeby.etherscan.io/address/0xaa88a3b70633c9746c5d258f402dae041e7f237a#code).

### CODE REVIEW

File: script/Deploy.sol [hunk 1]
```diff
@@ -0,0 +1,21 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "forge-std/Script.sol";
+import "./ValidatedAddress.sol";
+import "../src/CoWSwapEthFlow.sol";
+
+/// @title Deployer Sscript for the ETH Flow Contract
```
Comment (josojo) [inline script/Deploy.sol:8]: nit sscript

**Review by josojo — APPROVED** [review]
Cool.

I have not seen a "Deploy" contract like this before. But makes perfect sense. Cool!

File: script/ValidatedAddress.sol [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            "Settlement contract not available on this chain"
+        );
+        settlement = ICoWSwapSettlement(
+            0x9008D19f58AAbD9eD0D60971565AA8510560ab41
+        );
+        require(
+            CoWSwapEip712.domainSeparator(address(settlement)) ==
+                WithDomainSeparator(address(settlement)).domainSeparator(),
+            "Bad domain separator for settlement contract"
+        );
```
Comment (nlordell) [inline script/ValidatedAddress.sol:37]: This is cool - you can do some deploy-time assertions

**Review by nlordell — APPROVED** [review]
LGTM.

**Pull request merged**